### ISSUE TITLE
Unable to load editor on mobile

### ISSUE THREAD

**Issue by julianfairfax** [standardnotes/mobile#604]
I'm using the [Rich Markdown Editor](https://github.com/arturolinares/sn-rme) on my laptop, where it works fine, but when I open a note that was written with it on my phone, it displays an error saying `Unable to load Rich Markdown Editor - An unknown error occurred.`

**Comment by Tellervo89** [comment]
same issue on my Android phone, only occurs with custom themes though

**Comment by julianfairfax** [comment]
> same issue on my Android phone, only occurs with custom themes though

So this extension works for you? Or have you just only tried custom themes?

**Comment by Tellervo89** [comment]
rich markdown editor works on my laptop with my own custom theme. On my phone, rich markdown editor does not work with the same theme, and only with preinstalled themes from Standard Notes.

**Comment by julianfairfax** [comment]
> Hey @julianfairfax, thanks for reporting.
> 
> Have you tried reinstalling the editor and see if that fixes the issue?

I haven't actually. I will tomorrow when I get back to my laptop and I'll let you know.

**Comment by julianfairfax** [comment]
> Hey @julianfairfax, thanks for reporting.
> 
> Have you tried reinstalling the editor and see if that fixes the issue?

I have now. It didn't work. I'm on the 3.20 beta

**Comment by julianfairfax** [comment]
Can someone else test this on their device to confirm it's not just an issue for me?

**Comment by mariobrostech** [comment]
I experienced this too, but I was able to fix it with some interesting steps:
1) Create the note using the Rich Markdown Editor on my computer.
2) Open the note on mobile, and observe the note failing to open with the same error message as @julianfairfax reported in this issue.
3) Go to the main notes screen, and lock and unlock the device's screen (this was actually done as a mistake when I was trying to take a screenshot of the error, but I digress).
4) Observe the note loading perfectly fine when re-entering it. From this point on, every note created with the Rich Markdown Editor works just fine on the first try.
5) This can be replicated both on Android and iOS.
Not sure how helpful this is, especially considering that I'm unable to replicate the issue again, but it was certainly interesting to see how I was able to accidentally fix this.

**Comment by julianfairfax** [comment]
> I experienced this too, but I was able to fix it with some interesting steps:
> 
>     1. Create the note using the Rich Markdown Editor on my computer.
> 
>     2. Open the note on mobile, and observe the note failing to open with the same error message as @julianfairfax reported in this issue.
> 
>     3. Go to the main notes screen, and lock and unlock the device's screen (this was actually done as a mistake when I was trying to take a screenshot of the error, but I digress).
> 
>     4. Observe the note loading perfectly fine when re-entering it. From this point on, every note created with the Rich Markdown Editor works just fine on the first try.
> 
>     5. This can be replicated both on Android and iOS.
> 
> 
> Not sure how helpful this is, especially considering that I'm unable to replicate the issue again, but it was certainly interesting to see how I was able to accidentally fix this.

I have tried to replicate this fix multiple times, but it doesn't work for me. I hope this'll help you find a proper way to fix it though.

**Comment by Tellervo89** [comment]
> I experienced this too, but I was able to fix it with some interesting steps:
> 
>     1. Create the note using the Rich Markdown Editor on my computer.
> 
>     2. Open the note on mobile, and observe the note failing to open with the same error message as @julianfairfax reported in this issue.
> 
>     3. Go to the main notes screen, and lock and unlock the device's screen (this was actually done as a mistake when I was trying to take a screenshot of the error, but I digress).
> 
>     4. Observe the note loading perfectly fine when re-entering it. From this point on, every note created with the Rich Markdown Editor works just fine on the first try.
> 
>     5. This can be replicated both on Android and iOS.
> 
> 
> Not sure how helpful this is, especially considering that I'm unable to replicate the issue again, but it was certainly interesting to see how I was able to accidentally fix this.

Did not work for me either. However, since I updated Standard Notes to version 3.20.1 that was released on May 22, 22 on F-Droid, all editors load fine with custom themes without me having to do anything at all.

**Comment by julianfairfax** [comment]
> > I experienced this too, but I was able to fix it with some interesting steps:
> > 
> >     1. Create the note using the Rich Markdown Editor on my computer.
> > 
> >     2. Open the note on mobile, and observe the note failing to open with the same error message as @julianfairfax reported in this issue.
> > 
> >     3. Go to the main notes screen, and lock and unlock the device's screen (this was actually done as a mistake when I was trying to take a screenshot of the error, but I digress).
> > 
> >     4. Observe the note loading perfectly fine when re-entering it. From this point on, every note created with the Rich Markdown Editor works just fine on the first try.
> > 
> >     5. This can be replicated both on Android and iOS.
> > 
> > 
> > Not sure how helpful this is, especially considering that I'm unable to replicate the issue again, but it was certainly interesting to see how I was able to accidentally fix this.
> 
> Did not work for me either. However, since I updated Standard Notes to version 3.20.1 that was released on May 22, 22 on F-Droid, all editors load fine with custom themes without me having to do anything at all.  

This doesn't have anything to do with a custom theme. By all editors do you mean this one also works?

**Comment by Tellervo89** [comment]
Yes, the Rich Markdown Editor works. I assumed it was related to custom themes as all Standard Notes themes (e.g., solarized dark or midnight) loaded fine, and the editor only failed to load on my own themes.

**Comment by julianfairfax** [comment]
> > I experienced this too, but I was able to fix it with some interesting steps:
> > ```
> > 1. Create the note using the Rich Markdown Editor on my computer.
> > 
> > 2. Open the note on mobile, and observe the note failing to open with the same error message as @julianfairfax reported in this issue.
> > 
> > 3. Go to the main notes screen, and lock and unlock the device's screen (this was actually done as a mistake when I was trying to take a screenshot of the error, but I digress).
> > 
> > 4. Observe the note loading perfectly fine when re-entering it. From this point on, every note created with the Rich Markdown Editor works just fine on the first try.
> > 
> > 5. This can be replicated both on Android and iOS.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Not sure how helpful this is, especially considering that I'm unable to replicate the issue again, but it was certainly interesting to see how I was able to accidentally fix this.
> 
> I have tried to replicate this fix multiple times, but it doesn't work for me. I hope this'll help you find a proper way to fix it though.

I even tried to replicate this fix on a brand new account, but it still didn't work.

**Comment by julianfairfax** [comment]
> Hey @julianfairfax, thanks for reporting.
> 
> Have you tried reinstalling the editor and see if that fixes the issue?

Have you or any other developer tried to or been able to replicate this issue? I doubt it's just me since I tried that new account, but no one else seems to be complaining I don't know what's wrong but I'd really appreciate any insight whatsoever as Standard Notes is entirely useless to me on mobile at this point.

**Comment by julianfairfax** [comment]
> As a workaround you can remove the download_url property from the ext.json payload. The issue has to do with the offline editor capability of some editors. Still need to distill why.

I did here: https://raw.githubusercontent.com/julianfairfax/sn-rme/master/ext.json, but now it doesn't seem to install

**Comment by moughxyz** [comment]
I believe you prepend this to your note before publishing:

```
---
metatype: json
---
```

**Comment by julianfairfax** [comment]
> As a workaround you can remove the download_url property from the ext.json payload. The issue has to do with the offline editor capability of some editors. Still need to distill why.

It works! https://listed.to/p/6zv6GtamsT

**Comment by julianfairfax** [comment]
> As a workaround you can remove the download_url property from the ext.json payload. The issue has to do with the offline editor capability of some editors. Still need to distill why.

This means I can't edit my messages with this editor if I'm offline though.
